llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -402,6 +403,42 @@ Expected<GroupRef> GroupRef::create(MCCASBuilder &MB,
   return get(B->build());
 }
 
+static Expected<uint64_t>
+materializeDebugInfoSection(MCCASReader &Reader,
+                            const SmallVector<cas::ObjectRef> &Refs,
+                            DebugInfoSectionRef &DebugInfoRef) {
+  // Find DebugAbbrevSectionRef and get it's section contents
+  SmallVector<char> AbbrevContents;
+  for (auto ID : Refs) {
+    auto Node = Reader.getObjectProxy(ID);
+    if (!Node)
+      return Node.takeError();
+    if (auto F = DebugAbbrevSectionRef::Cast(*Node)) {
+      StringRef Remaining = F->getData();
+      auto AbbrevRefs = decodeReferences(*F, Remaining);
+      if (!AbbrevRefs)
+        return AbbrevRefs.takeError();
+      for (auto AbbrevID : *AbbrevRefs) {
```

Comment:
This is very similar to what `LoadedDebugInfoSection` does.
IMO it would be better if we had a `LoadedAbbrevSection`, and this entire method could be:

```cpp
tatic Expected<uint64_t>
 materializeDebugInfoSection(MCCASReader &Reader,
                             const SmallVector<cas::ObjectRef> &Refs,
                             DebugInfoSectionRef &DebugInfoRef) {
   auto AbbrevSectionRef = findAbbrevSection(Refs);
   // check for errors
   auto LoadedAbbrev = LoadedAbbrevSection::load(*AbbrevSectionRef);
   // check for errors
   return DebugInfoRef.materialize(Reader, std::move(*LoadedAbbrev));
}
```

Reply:
We currently don't check whether `PaddingRef` is well formed, i.e. a single one and always at the end.

Reply:
It's difficult to see, but there is currently no error handling in case we fail to find the AbbrevRef, with the suggestion above it becomes clear that we there will be error handling

Reply:
Though tbf, the way it is written now, we do not check if the PaddingRef is the last node in the DebugInfoSectionRef either

Reply:
We do!

<img width="568" alt="image" src="https://user-images.githubusercontent.com/5406686/191305398-77850d59-1021-4d19-9006-4ea941dcd267.png">

Reply:
nvm we do 😅